### PR TITLE
Adds basic documentation and example for external access integrations & secrets (Snowpark concept)

### PR DESCRIPTION
**Pull request by ernestoongaro**
## What are you changing in this pull request and why?
https://github.com/dbt-labs/dbt-snowflake/issues/925
https://github.com/dbt-labs/dbt-snowflake/pull/955

## Checklist
Wait until PR is merged

### CODE REVIEW

File: website/docs/docs/build/python-models.md [hunk 1]
```diff
@@ -630,6 +630,26 @@ def model(dbt, session):
     )
 ```
 
+**External Access Integrations and Secrets**
+If you want to use Snowflake’s [external access]([url](https://docs.snowflake.com/en/developer-guide/external-network-access/external-network-access-overview)) together with [secrets]([url](https://docs.snowflake.com/en/developer-guide/external-network-access/secret-api-reference)) to query external APIs within dbt Python models there are some additional configurations to use:
```
Comment (mirnawong1) [inline website/docs/docs/build/python-models.md]:
```suggestion
To query external APIs within dbt Python models, use Snowflake’s [external access](https://docs.snowflake.com/en/developer-guide/external-network-access/external-network-access-overview) together with [secrets](https://docs.snowflake.com/en/developer-guide/external-network-access/secret-api-reference). Here are some additional configurations you can use:
```

File: website/docs/docs/build/python-models.md [hunk 2]
```diff
@@ -630,6 +630,26 @@ def model(dbt, session):
     )
 ```
 
+**External Access Integrations and Secrets**
```
Comment (mirnawong1) [inline website/docs/docs/build/python-models.md]:
```suggestion
**External access integrations and secrets**
```

**Comment by mirnawong1** [comment]
going to cc @matthewshaver  as it looks like this related to v1.8. have mad some suggestions and the linked snowflake prs are still open as of today (4/11)

**Comment by ernestoongaro** [comment]
tysm @mirnawong1 for your edits!

**Review by mirnawong1 — APPROVED** [review]
🚢 lgtm and thank you so v much @ernestoongaro for the initiative ✨

**Comment by mirnawong1** [comment]
fixing spacing bc the rendering is looking weird
Screenshot 2024-04-12 at 16 31 49

**Pull request merged**